scripts/lib/api/generateToc.test.ts

```diff
@@ -148,10 +149,4 @@ const pkg = {
   versionWithoutPatch: "1.0",
   releaseNoteEntries: [],
   historical: false,
-
-  // These properties are not used but included to appease typescript
-  githubSlug: "qiskit/qiskit-ibm-runtime",
-  baseUrl: `https://qiskit.org/ecosystem/ibm-runtime`,
-  initialUrl: `https://qiskit.org/ecosystem/ibm-runtime/apidocs/ibm-runtime.html`,
-  transformLink: undefined,
-};
+} as unknown as Pkg;
```

Comment:
Idk why TypeScript made me do this `as unknown` here but not in another test.

Reply:
Maybe you need `GithubSlug`?

Reply:
Yeah, that's what the error message was saying. But to clarify, I don't want to have to specify irrelevant types here which IMO makes the test harder to read. We get around this here:

https://github.com/Qiskit/documentation/blob/1ae2fd56e55573395897326ecb3913839d456c4c/scripts/lib/api/addFrontMatter.test.ts#L56-L58

Reply:
Making `GithubSlug` optional allows us to drop the cast to unknown, but maybe it doesn't make sense because all the projects are required to have it, right?
```ts
export type PkgInfo = {
  name: string;
  githubSlug?: string;
```

Reply:
Hmm... or maybe the type in the function?

```ts
(): Pkg => {
```

Reply:
I don't want to make it optional because it's important that we set `githubSlug` in production. Instead, I'm planning to make `Pkg` a proper class in a follow up PR and to have a static constructor like `Pkg.test()` that uses default arts for every option so that tests are easier for us to work with and we avoid these "incomplete types".

Reply:
The constructor is a good idea